AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -85,25 +87,94 @@ void ASDisplayNodeRespectThreadAffinityOfNode(ASDisplayNode *node, void (^block)
   }
 }
 
-+ (void)initialize
-{
-  if (self == [ASDisplayNode class]) {
-    return;
+/**
+ *  Returns ASDisplayNodeFlags for the givern class/instance. instance MAY BE NIL.
+ *
+ *  @param c        the class, required
+ *  @param instance the instance, which may be nil. (If so, the class is inspected instead)
+ *
+ *  @return ASDisplayNode flags.
+ */
+static struct ASDisplayNodeFlags GetASDisplayNodeFlags(Class c, ASDisplayNode *instance) {
+  struct ASDisplayNodeFlags flags = {0};
+
+  flags.isInHierarchy = NO;
+  flags.displaysAsynchronously = YES;
```

Comment:
I'll add a nil check for c just to be safe. (I didn't bother since I made it a static function and knew the context where it could be called.)

The values here are only defaults that were originally set in _initializeInstance and then overridden in either a subclasses' init method or after _initializeInstance runs (as in initWithViewBlock/LayerBlock) so I think we are good there.